File: Leetcode daily/14_oct.cpp
#include<bits/stdc++.h>
using namespace std;

class Solution {
public:
    ListNode* deleteMiddle(ListNode* head) {
        ListNode* slow = head, *fast = head;
        ListNode *newNode = new ListNode(-1,head);
        ListNode* prev = newNode;
        while(fast and fast->next){
            prev = slow;
            slow = slow->next;
            fast = fast->next->next;
        }
        prev->next = slow->next;
        return newNode->next;
    }
};

File: Leetcode daily/14Oct.java
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        if(head.next==null){
            return null;
        }
        ListNode fast=head,slow=head,prev=null;
        while(fast!=null && fast.next!=null){
            fast=fast.next.next;
            prev=slow;
            slow=slow.next;
        }
        prev.next=prev.next.next;
        return head;
    }
}
